### PR TITLE
schildichat: change livecheck strategy

### PR DESCRIPTION
**Pull request by daeho-ro**
Latest version doesn't have the macOS .dmg file and so changed the livecheck strategy to the github releases with asset checking.

(comment moved from bottom to top)

----

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

File: Casks/s/schildichat.rb [hunk 1]
```diff
@@ -10,8 +10,19 @@
...
+        next if release["test"]
+
+        release["assets"]&.map do |asset|
+          match = asset["name"]&.match(regex)
+          next if match.blank?
+
+          match[1]
+        end
+      end.flatten
+    end
```
Comment (chenrui333) [inline Casks/s/schildichat.rb:25]: why this change?

Comment (chenrui333) [reply Casks/s/schildichat.rb:25]: Oh, I see the point. (sounds like upstream issue)

---

if they stop publishing the universal dmg artifacts for the new releases, maybe better to deprecate to notify the users.

**Review by bevanjkay — DISMISSED** [review]
Based on the response from upstream, looks like MacOS builds are discontinued.

**Comment by daeho-ro** [comment]
I will change when I available.

**Review by krehel — APPROVED** [review]
